lib/enumerize/activerecord.rb

```diff
@@ -21,8 +21,14 @@ def enumerize(name, options={})
           require 'enumerize/hooks/uniqueness'
 
           unless options[:multiple]
-            decorate_attribute_type(name, :enumerize) do |subtype|
-              Type.new(enumerized_attributes[name], subtype)
+            if method(:decorate_attribute_type).parameters.count == 2
```

Comment:
shouldn't it be the other way around? When `parameters.count == 2` then use `decorate_attribute_type(name, :enumerize)`? 

Anyway I think it' better to use Rails version check there to decide which method to use. In that case it's clear that the difference is about Rails version. What do you think?

Reply:
`decorate attribute_type` accepts other parameters(positional arguments and block argument) per versions. `Method#parameters` is as follows.

**Rails 6.1**    

```ruby
def decorate_attribute_type(attr_name, **default)  
#=> [[:req, :attr_name], [:keyrest, :default]]
``` 

**Rails 6.0**   

```ruby
def decorate_attribute_type(column_name, decorator_name, &block)  
#=> [[:req, :column_name], [:req, :decorator_name], [:block, :block]] 
``` 

So the current check is correct. 

I think the behavior check is more accurate than the version check. But this checking seems a little confusing.So I will update this to check the Rails version.

Reply:
@nashby  Done!